### ISSUE TITLE
`ListFormatStyle` missing

### ISSUE THREAD

**Issue by 0xTim** [apple/swift-corelibs-foundation#4596]
The Objective-C Foundation has a `ListFormatStyle` for formatting collections as described at https://developer.apple.com/documentation/foundation/listformatstyle. This does not exist in Swift Foundation.

Expected Result:

Using `someArrayOfStrings.formatted(.list(type: .and))` works on both macOS and Linux

Actual Result:

```
.../Something.swift:151:46: error: value of type '[String]' has no member 'formatted'
let formatted = someArrayOfStrings.formatted(.list(type: .and))
```

Steps to reproduce

```swift
let someArrayOfStrings = ["Alice", "Bob", "Charlie"]
let formatted = someArrayOfStrings.formatted(.list(type: .and))
```

**Comment by tomerd** [comment]
cc @millenomi

**Comment by RandomHashTags** [comment]
same error for `URLSession`'s
```swift
public func data(for request: URLRequest, delegate: (URLSessionTaskDelegate)? = nil) async throws -> (Data, URLResponse)
```
function.

**Comment by drewster99** [comment]
> same error for `URLSession`'s
> 
> ```swift
> public func data(for request: URLRequest, delegate: (URLSessionTaskDelegate)? = nil) async throws -> (Data, URLResponse)
> ```
> 
> function.

Probably the URLSession one should be a separate Issue.

**Comment by 0xTim** [comment]
That's being tracked at https://github.com/apple/swift-corelibs-foundation/issues/3205

**Comment by parkera** [comment]
This is available in swift-foundation, so resolved by #5001
